src/plain.jl

```diff
@@ -14,6 +14,12 @@ import Base: close, convert, done, dump, eltype, endof, flush, getindex,
 
 include("datafile.jl")
 
+if VERSION < v"0.4.0-dev+4319"
+    getnames(T::DataType) = T.names
+else
+    getnames(T::DataType) = T.name.names
+end
```

Comment:
There is a `fieldnames` function in 0.4 (and Compat) that you can use instead of defining this. `fieldnames` also appears to work on some types where this would not, e.g. DataType itself.